pkg/kubectl/cmd/util/openapi/validation/validation.go

```diff
@@ -118,11 +119,14 @@ func getObjectKind(object interface{}) (schema.GroupVersionKind, error) {
 		listErrors = append(listErrors, errors.New("kind isn't string type"))
 	}
 	if listErrors != nil{
```

Comment:
Can we move this to the end? I know we can't because we don't know if `apiVersion` is a string. I'll explain in a different comment.

Reply:
@apelisse +1 and Yes, that's the reason I didn't move but I see that the need and found easier way of doing it. So I actually tried to take care of it in the new commit I just uploaded but please let me know if I didn't do it the way you are thinking :-). Thanks!